### PR TITLE
Add insert_into_stream_table functions

### PR DESCRIPTION
**Pull request by acco**

### CODE REVIEW

**Comment by acco** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/93?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#93** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/93?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#84** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/84?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 6 other dependent PRs ([#85](https://github.com/sequinstream/sequin/pull/85) <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/85?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#86](https://github.com/sequinstream/sequin/pull/86) <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/86?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#87](https://github.com/sequinstream/sequin/pull/87) <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/87?utm_source=stack-comment-icon" target="_blank">Graphite</a> and 3 others)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @acco and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: server/test/support/factory/streams_factory.ex [hunk 1]
```diff
@@ -367,6 +367,19 @@ defmodule Sequin.Factory.StreamsFactory do
     |> Repo.insert!()
   end
 
+  def stream_table_record(attrs \\ []) do
+    merge_attributes(
+      %{
```
Comment (RTLS) [inline server/test/support/factory/streams_factory.ex:372]: should use NewMessage struct ?

File: server/lib/sequin/streams/stream_table.ex [hunk 2]
```diff
@@ -118,6 +132,11 @@ defmodule Sequin.Streams.StreamTable do
     end
   end
 
+  def all_column_names(%__MODULE__{} = stream_table) do
+    Enum.map(base_column_names(), &Atom.to_string/1) ++
+      Enum.map(stream_table.columns, & &1.name)
+  end
```
Comment (RTLS) [inline server/lib/sequin/streams/stream_table.ex:138]: might want to preload here (which is a no op if columns already preloaded) ?

File: server/lib/sequin/streams/streams.ex [hunk 3]
```diff
@@ -872,4 +873,48 @@ defmodule Sequin.Streams do
...
+      |> Enum.reject(&(&1 in conflict_target or &1 == "sequin_id"))
+
+    Repo.insert_all(
+      stream_table.table_name,
+      records_for_insert(stream_table.columns, records),
+      conflict_target: conflict_target,
+      on_conflict: {:replace, replace},
+      prefix: stream_table.table_schema_name
+    )
+  end
```
Comment (RTLS) [inline server/lib/sequin/streams/streams.ex:905]: wow so chill ! Ecto 😍

File: server/lib/sequin/streams/streams.ex [hunk 4]
```diff
@@ -872,4 +873,48 @@ defmodule Sequin.Streams do
...
+
+  defp records_for_insert(columns, records) do
+    Enum.map(records, fn record ->
+      record = Map.update!(record, "sequin_id", &UUID.string_to_binary!/1)
+
+      Enum.reduce(columns, record, fn %StreamTableColumn{} = column, record ->
```
Comment (RTLS) [inline server/lib/sequin/streams/streams.ex:911]: we should use some struct for records- we have NewMessage for now

File: server/test/support/factory/streams_factory.ex [hunk 5]
```diff
@@ -367,6 +367,19 @@ defmodule Sequin.Factory.StreamsFactory do
...
+        sequin_id: Ecto.UUID.generate(),
+        seq: Enum.random(1..1_000_000),
+        data: Jason.encode!(%{random_key: "random_value"}),
```
Comment (RTLS) [inline server/test/support/factory/streams_factory.ex:375]:
```suggestion
        data: Jason.encode!(%{Factory.Lorem.word() => Factory.Lorem.word()}),
```

perhaps?

File: server/test/support/factory/streams_factory.ex [hunk 6]
```diff
@@ -367,6 +367,19 @@ defmodule Sequin.Factory.StreamsFactory do
     |> Repo.insert!()
   end
 
+  def stream_table_record(attrs \\ []) do
+    merge_attributes(
+      %{
+        sequin_id: Ecto.UUID.generate(),
+        seq: Enum.random(1..1_000_000),
```
Comment (RTLS) [inline server/test/support/factory/streams_factory.ex:374]: unique positive integer?

File: server/test/support/factory/streams_factory.ex [hunk 7]
```diff
@@ -367,6 +367,19 @@ defmodule Sequin.Factory.StreamsFactory do
...
+        data: Jason.encode!(%{random_key: "random_value"}),
+        recorded_at: DateTime.truncate(DateTime.utc_now(), :second),
```
Comment (RTLS) [inline server/test/support/factory/streams_factory.ex:376]: why is this truncated?

**Review by RTLS — APPROVED** [review]

**Pull request closed without merging**